pkg/asset/installconfig/azure/session.go

```diff
@@ -30,6 +32,7 @@ var (
 type Session struct {
 	GraphAuthorizer autorest.Authorizer
 	Authorizer      autorest.Authorizer
+	RequestAdapter  *msgraphsdk.GraphRequestAdapter
```

Comment:
The problem is that to create a `GraphRequestAdapter` I need an `AzureIdentityAuthenticationProvider` which in turn needs an `azidentity.ClientSecretCredential`. So either I save in the session:
1) the request adapter itself,
2) the authentication provider,
3) the client secret credential.
Which one would you rather pick, 2 or 3?

Reply:
3 because it is common across (the new) clients.

Reply:
I removed it from the session altogether since it's only used in one place for now. When more places start using it we can share it through the session.